### DISCUSSION TITLE
in class component "this.props.match.params" is not working in 6.0.0 react-route-dom

### DISCUSSION THREAD

**Discussion by rahulummadisetti** [remix-run/react-router#8404] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/88239772/143769993-c585ce46-3cc0-4e32-9fe4-04591411b1f2.png)

in class component "this.props.match.params" is not working in 6.0.0 react-route-dom

import React,{ Component } from 'react';

class Customer extends Component{
    constructor(props){
        super();
    }
    render() {
        
        console.log(this.props);
        const  { params }  = this.props.match;

        return (
            <div>
            <h1>Customer</h1>
            <p>{params.id}</p>
            </div>
        )
    }
}

export default Customer;

**Comment by Manikandan-Raju** [comment]
running into same issue. do u have any solution?

**Comment by timdorr** [comment, marked as answer]
This isn't a feature in v6. You need to use hooks or use a wrapper around your class components to provide the hook data to them.

**Comment by Dricc-new** [comment]
On  [this page](https://stackoverflow.com/questions/69967745/react-router-v6-access-a-url-parameter) you will find the solution that, although a bit long, worked for me with react-router-dom 6.14.2

**Comment by Nazir-Rizwan** [comment]
`useParams ` use for match url route path

**Comment by Naim637** [comment]
Just watch this 9-minute [tutorial](https://youtu.be/y_pr4lRoUto?feature=shared).

**Reply by Jogata** [reply]
The question is "in class component" - this video is for function component.
